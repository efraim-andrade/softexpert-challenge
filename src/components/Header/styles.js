import styled from 'styled-components';
import { Link } from 'react-router-dom';
import media from 'styled-media-query';

import { fonts, colors, metrics } from '~/theme';

export const Container = styled.div`
  margin-bottom: ${metrics.baseSpacing(2)};

  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Left = styled.div`
  width: 100%;

  > h2 {
    ${fonts.size3};
    color: ${colors.secondary};
    opacity: 0.6;

    ${media.lessThan('medium')`
      ${fonts.size1};
    `}
  }

  > h1 {
    max-width: 90%;

    ${fonts.size4};
    color: ${colors.black};

    ${media.lessThan('medium')`
      ${fonts.size2};
    `}
  }
`;

export const Right = styled(Link)`
  display: flex;
  align-items: center;
  justify-content: center;

  cursor: pointer;

  ${media.lessThan('medium')`
    flex-direction: column;
  `}

  &:hover {
    > svg {
      transform: translateX(-10%);
    }
  }

  &:visited {
    color: ${colors.black};
  }

  > svg {
    margin-right: ${metrics.baseSpacing()};

    transition: 0.3s;

    ${media.lessThan('medium')`
      margin-right: 0;
    `}
  }

  > span {
    ${media.lessThan('medium')`
      ${fonts.size0};
    `}
  }
`;
