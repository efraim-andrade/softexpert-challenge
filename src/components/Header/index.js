import React from 'react';
import PropTypes from 'prop-types';
import { ArrowLeftCircle } from 'styled-icons/feather/ArrowLeftCircle';

import { Container, Left, Right } from './styles';

export default function Header({ title, subtitle }) {
  return (
    <Container>
      <Left>
        <h2>{subtitle}</h2>

        <h1>{title}</h1>
      </Left>

      <Right to="/">
        <ArrowLeftCircle size="32" />

        <span>Voltar</span>
      </Right>
    </Container>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};
