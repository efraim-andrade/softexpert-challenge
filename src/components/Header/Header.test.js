import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';

import Header from './index';

describe('Components - Header', () => {
  const history = createMemoryHistory();

  const props = {
    subtitle: 'SoftExpert',
    title: 'Stock Exchange Challenge',
  };

  it('should render correctly', () => {
    const { asFragment } = render(<Header {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the goback button when it's not in the home", () => {
    window.history.pushState({}, '', '/details');

    const { getByTestId } = render(
      <Router history={history}>
        <Header {...props} />
      </Router>
    );

    const goBackBtn = getByTestId('goBack');

    expect(goBackBtn).toBeDefined();
  });
});
