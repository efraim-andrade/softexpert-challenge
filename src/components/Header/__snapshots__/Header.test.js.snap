// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Header should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bwzfXH hslxvQ"
  >
    <div
      class="sc-bxivhb fHeAsf"
    >
      <h2>
        SoftExpert
      </h2>
      <h1>
        Stock Exchange Challenge
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
